app/src/main/kotlin/com/flxrs/dankchat/utils/extensions/ColorExtensions.kt

```diff
@@ -44,4 +45,22 @@ fun Int.normalizeColor(@ColorInt background: Int): Int {
 }
 
 @ColorInt
-fun Int.harmonize(context: Context): Int = MaterialColors.harmonizeWithPrimary(context, this)
\ No newline at end of file
+fun Int.harmonize(context: Context): Int = MaterialColors.harmonizeWithPrimary(context, this)
+
+
+/** helper to extract only RGB part (i.e. drop the alpha part) */
+fun Int.onlyRGB(): Int = this and 0xffffff
+
+/** convert int to RGB with zero pad */
+fun Int.toHexCode(): String = Integer.toHexString(this.onlyRGB()).padStart(6, '0')
+
+/** convert RGB color (0xffffff) to ARGB with alpha */
+fun Int.toARGBInt(alpha: Int = 255): Int = (alpha shl 24) or this.onlyRGB()
+
+
+/** find best contrast text to display on specified background color (whtie or black) -- opacity ignored */
+fun Int.getContrastTextColor(): Int {
+    val possibleColors = listOf(Color.BLACK, Color.WHITE)
```

Comment:
Keep in mind that the regular text colors are not black or white, instead `R.attr.colorOnSurface` is used , depending on the current theme and the inverse `colorOnSurface`. So I would instead like to see those colors being used for the current theme.
Also instead of this hacky approach you can probably just use `MaterialColors.isColorLight`